### DISCUSSION TITLE
What is the policy for unique IDs with respect to "generic" integrations?

### DISCUSSION THREAD

**Discussion by fschild** [home-assistant/architecture#702] [General] [answered]
There are dozens of integrations which will never fulfill the Unique ID requirements outlined in
[Entity Registry | Home Assistant Developer Docs.](https://developers.home-assistant.io/docs/entity_registry_index/).

I am talking about integrations such as:

    MQTT
    template
    SQL
    RESTful*
    ping
    command line

and many more specific integrations which cannot pull enough identity data from their respective API.

Currently, some of these, such as MQTT and template, allow the creation of unique_ids in the configuration.yaml file, others such as SQL, rest and ping do not allow it.

This is extremely inconsistent!

Given that the missing unique IDs make the handling of these integrations in Lovelace very difficult and a lot of users complain about it, what should the roadmap be?

1. Allow (and implement) the manual setting of unique_ids for these integrations?
2. Document it clearly to users that they have to feed data from these integrations into a template integration to make them usable in Lovelace?

**Comment by frenck** [comment, marked as answer]
This has been defined by ADR-0010.

If ADR-0010 applies to an integration, those integrations can be able to provide a unique ID in almost all cases (using the config entry ID as a fallback).

In all other cases, we allow for YAML configuration of it. That said, even in those cases, an actual unique ID (non-manual) is preferred.

../Frenck
